src/Collector/InterfaceNameRegexCollector.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Qossmic\Deptrac\Collector;
+
+use LogicException;
+use Qossmic\Deptrac\AstRunner\AstMap;
+use Qossmic\Deptrac\AstRunner\AstMap\AstClassReference;
+use Qossmic\Deptrac\AstRunner\AstMap\ClassLikeName;
+
+class InterfaceNameRegexCollector extends RegexCollector implements CollectorInterface
+{
+    public function getType(): string
+    {
+       return 'interfaceNameRegex';
+    }
+
+    public function resolvable(array $configuration, Registry $collectorRegistry, array $resolutionTable): bool
+    {
+        return true;
+    }
+
+    public function satisfy(
+        array $configuration,
+        AstMap\AstTokenReference $astTokenReference,
+        AstMap $astMap,
+        Registry $collectorRegistry,
+        array $resolutionTable = []
+    ): bool {
+        if (!$astTokenReference instanceof AstMap\AstClassReference) {
+            return false;
+        }
+
+        $interfaceName = $this->getPattern($configuration);
+
+        $reflection  = (new \ReflectionClass($astTokenReference->getTokenName()->toString()));
```

Comment:
@dbrumann Is there an option to get this information (object type e.g. interface or class) with php-parser instead?

Reply:
Yes there is, you just need to propagate it, as we do not collect that information at the moment (there was no need for it). Look at https://github.com/qossmic/deptrac/blob/main/src/AstRunner/AstParser/NikicPhpParser/FileReferenceVisitor.php#L249, where the `$node` has the information about it being an interface or not.